### PR TITLE
search: prevent the rendering of the home page when navigating back to the search results

### PR DESCRIPTION
**Pull request by valerybugakov**
### TLDR

We render the search home page when navigating back to search results from the blob view. Don't believe me? 
Go to [the cloud instance](https://sourcegraph.com/search), set a breakpoint in the `SearchPage.tsx` file in dev tools and try yourself!

### Details

Currently, we have `parsedSearchQuery` in the state of the `SourcegraphWebApp` component. 
1. Upon opening search results — its value is updated with the current search query. 
2. After opening any search result — its value is set to an empty string.
3. When we navigate back to search results we there's a delay in updating its value because:
    - The update is done in the `useEffect` hook in the `Layout` component, which is rendered after we figure out which React router route to use.
https://github.com/sourcegraph/sourcegraph/blob/ee1e1fe5af79de620292c6794dcbff2c6a47fe71/client/web/src/Layout.tsx#L155-L158
    - It means that the code responsible for deciding what to render doesn't have up-to-date `parsedSearchQuery` available on the initial render. It leads to rendering the `SearchPage` component, which we don't need, negatively affecting the perceived performance.
https://github.com/sourcegraph/sourcegraph/blob/ee1e1fe5af79de620292c6794dcbff2c6a47fe71/client/web/src/routes.tsx#L78-L82

## Changes

- This PR removes `parsedSearchQuery` from the state of the `SourcegraphWebApp` component.
    - It seems it's required only in the `SourcegraphWebApp` `componentDidMount` hook, and deleting it from the state doesn't break things.
- Uses `parsedSearchQuery` directly from the `Layout` component that we already have in place.

As a result — the `StreamingSearchResults` component is rendered right away when navigating back to search results without redundant intermediate steps.

Preparation to address https://github.com/sourcegraph/sourcegraph/issues/28845.

### CODE REVIEW

**Review by vovakulikov — APPROVED** [review]
LGTM. Our root components (`SourcegraphWebApp.tsx` and `Layout.tsx`) could be much better I think and easy to change if we removed all this legacy top-level state from it. @sourcegraph/search-core seems like most of this state is search specific. Maybe you have thoughts about how can we move it safely from root components to consumers or move it in our Zustand stores if this state has too many consumers.

**Comment by fkling** [comment]
@vovakulikov Yeah, I'm in the process of moving all that state over to `navbarQueryState`.

**Review by limitedmage — APPROVED** [review]

**Review by novoselrok — APPROVED** [review]
